### PR TITLE
Monday morning pull request

### PR DESCRIPTION
**Pull request by wardymate**
Not finished as some tests failing

### CODE REVIEW

File: lib/array.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+			self.each do |item|
+				arg = yield arg , item
+			end
+			arg
+		end
+	end
+
+
+
+end
```
Comment (spike01) [inline lib/array.rb:17]: Indentation!

File: lib/array.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Array
+
+	def my_inject(arg=0)
+		if arg.is_a? Symbol
+			 proc = arg.to_proc
+			 self.my_inject(&proc)
```
Comment (spike01) [inline lib/array.rb:6]: Hah! Very neat. I actually had to fire up Pry to check this out

File: spec/array_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		expect(array.my_inject(:-)).to eq (-4)
+		expect(array.inject(:-)).to eq (-4)
+	end
+
+	xit 'should subtract each number from the previous' do
+		expect(array.my_inject(0){|start,number|start-number}).to eq (-4)
+		expect(array.inject(0){|start,number|start-number}).to eq (-4)
+	end
+
+end
```
Comment (spike01) [inline spec/array_spec.rb:49]: Consider storing the result of the real inject as a variable and test against that.

**Comment by oceansize** [comment]
Taking into account that this isn't finished, I would advise that you go back a few steps and concentrate on what the method returns. Make sure that you understand what happens with inject when there's an argument passed and when there is none. Only when that is working as the original should you worry about passing the symbols :)

**Pull request closed without merging**